app/controllers/posts_controller.rb

```diff
@@ -1,69 +1,19 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: %i[ show edit update destroy ]
 
-  # GET /posts or /posts.json
-  def index
-    @posts = Post.all
-  end
-
-  # GET /posts/1 or /posts/1.json
-  def show
-  end
-
-  # GET /posts/new
   def new
     @post = Post.new
   end
 
-  # GET /posts/1/edit
-  def edit
-  end
-
-  # POST /posts or /posts.json
   def create
-    @post = Post.new(post_params)
-
-    respond_to do |format|
-      if @post.save
-        format.html { redirect_to @post, notice: "Post was successfully created." }
-        format.json { render :show, status: :created, location: @post }
-      else
-        format.html { render :new, status: :unprocessable_entity }
-        format.json { render json: @post.errors, status: :unprocessable_entity }
-      end
+    if @post.invalid?
+      render "new"
+    else
+      render "show"
     end
-  end
 
-  # PATCH/PUT /posts/1 or /posts/1.json
-  def update
-    respond_to do |format|
-      if @post.update(post_params)
-        format.html { redirect_to @post, notice: "Post was successfully updated." }
-        format.json { render :show, status: :ok, location: @post }
-      else
-        format.html { render :edit, status: :unprocessable_entity }
-        format.json { render json: @post.errors, status: :unprocessable_entity }
-      end
-    end
   end
 
-  # DELETE /posts/1 or /posts/1.json
-  def destroy
-    @post.destroy
-    respond_to do |format|
-      format.html { redirect_to posts_url, notice: "Post was successfully destroyed." }
-      format.json { head :no_content }
-    end
+  def show
+    @post
```

Comment:
This will be null. The show action is connected to a route that has an `id` in the URL. To access the id you can use params. A typical show route will have something like this `@post = Post.find(params[:id])`